src/gateway/reference/proxy.md

```diff
@@ -1265,32 +1265,104 @@ Host: my-websocket-api.com
 Upgrade: WebSocket
 ```
 
-This will make Kong forward the `Connection` and `Upgrade` headers to your
+This makes {{site.base_gateway}} forward the `Connection` and `Upgrade` headers to your
 upstream service, instead of dismissing them due to the hop-by-hop nature of a
 standard HTTP proxy.
 
+### WebSocket proxy modes
+
+There are two methods for proxying WebSocket traffic in {{site.base_gateway}}:
+
+* HTTP(S) services and routes
+* WS(S) services and routes
+
+{% navtabs %}
+{% navtab HTTP(S) services and routes %}
+
+Services and routes using the `http` and `https` protocols are fully capable of
+handling WebSocket connections with no special configuration. With this method,
+WebSocket sessions behave identically to regular HTTP requests, and all of the
+request/response data is treated as an opaque stream of bytes.
+
+```yaml
+services:
+  - name: my-http-websocket-service
+    protocol: http
+    host: 1.2.3.4
+    port: 80
+    path: /
+    routes:
+      - name: my-http-websocket-route
+        protocols:
+          - http
+          - https
+```
+
+{% endnavtab %}
+
+{% navtab WS(S) services and routes %}
+{:.badge .enterprise}
```

Comment:
@cloudjumpercat when you have a sec, could you take a look and see if there's a better place for this enterprise badge? Right now it's not great:

![image](https://user-images.githubusercontent.com/3277009/181599685-069cfdd3-d116-4a48-8664-65153216b27e.png)

https://deploy-preview-4150--kongdocs.netlify.app/gateway/latest/reference/proxy/#proxy-websocket-traffic

My intent was to have the badge _in_ the navbar title for this (`WS(S) Services and Routes {{ENTERPRISE}}`).

Reply:
I'm honestly not 100% sold on the navbar for this section anyways, since it makes the page less friendly to `ctrl+f` (which is a shame for something in the "reference" category). Maybe I'll just go back to plain markdown sections.

Reply:
I decided to remove the navbar from this section.